keras_core/utils/layer_utils.py

```diff
@@ -0,0 +1,62 @@
+import numpy as np
+
+from keras_core.utils.module_utils import tensorflow as tf
+
+
+def validate_string_arg(
+    input_data,
+    allowable_strings,
+    layer_name,
+    arg_name,
+    allow_none=False,
+    allow_callables=False,
+):
+    """Validates the correctness of a string-based arg."""
+    if allow_none and input_data is None:
+        return
+    elif allow_callables and callable(input_data):
+        return
+    elif isinstance(input_data, str) and input_data in allowable_strings:
+        return
+    else:
+        allowed_args = "`None`, " if allow_none else ""
+        allowed_args += "a `Callable`, " if allow_callables else ""
+        allowed_args += f"or one of the following values: {allowable_strings}"
+        if allow_callables:
+            callable_note = (
+                f"If restoring a model and `{arg_name}` is a custom callable, "
+                "please ensure the callable is registered as a custom object. "
+                "registering_the_custom_object for details. "
+            )
+        else:
+            callable_note = ""
+        raise ValueError(
+            f"Unkown value for `{arg_name}` argument of layer {layer_name}. "
+            f"{callable_note}Allowed values are: {allowed_args}. Received: "
+            f"{input_data}"
+        )
+
+
+def listify_tensors(x):
+    """Convert any tensors or numpy arrays to lists for config serialization."""
+    if tf.is_tensor(x):
+        x = x.numpy()
+    if isinstance(x, np.ndarray):
+        x = x.tolist()
+    return x
+
+
+def ensure_tensor(inputs, dtype=None):
```

Comment:
You can inline this code where it is used

Reply:
Since it is not a simple piece, I think its better for it to be available as a function call. 🤔